src/hybrid_reader.c

```diff
@@ -169,6 +169,38 @@ void computeDistances(HybridIterator *hr) {
   IndexResult_Free(cur_vec_res);
 }
 
+// Review the estimated child results num, and returns true if hybrid policy should change.
+static bool reviewHybridSearchPolicy(HybridIterator *hr, size_t n_res_left, size_t child_upper_bound,
+                                     size_t *child_num_estimated) {
+  // Re-evaluate the child num estimated results and the hybrid policy based on the current batch.
+  size_t new_results_cur_batch = heap_count(hr->topResults) - (hr->query.k - n_res_left);
+  // This is the ratio between index_size to child results size as reflected by this batch.
+  float cur_ratio = (float)new_results_cur_batch / n_res_left;
+  // Child estimated number of results as reflected by this batch.
+  size_t cur_child_num_estimated = cur_ratio * VecSimIndex_IndexSize(hr->index);
+  // Conclude the new estimation of the child res num as the average between the old
+  // and new estimation (get the accumulated estimation).
+  *child_num_estimated = (*child_num_estimated + cur_child_num_estimated) / 2;
+  if (*child_num_estimated > child_upper_bound) {
+    *child_num_estimated = child_upper_bound;
+  }
+  if ((VecSimSearchMode)hr->runtimeParams.searchMode == VECSIM_HYBRID_BATCHES) {
+    return false;
+  }
```

Comment:
The previous check (line 258) is to see if both `SearchMode == VECSIM_HYBRID_BATCHES` and `batchSize` argument were given (because if so, we don't need to review and re-evaluate the batch size).
Here, we check if `SearchMode == VECSIM_HYBRID_BATCHES` (but `batchSize` was not given) - in that case, we don't call `VecSimIndex_PreferAdHocSearch` (as we definitely not going to change the policy). Otherwise, we do call ``VecSimIndex_PreferAdHocSearch` with the updated `child_num_estimated`, and check if we should switch to ad-hoc BF.